raylib_wrap.nim

```diff
@@ -0,0 +1,131 @@
+
+proc getMonitorName*(monitor: int32): string {.inline.} =
+  ## Get the human-readable, UTF-8 encoded name of the primary monitor
+  result = $getMonitorNamePriv(monitor)
+
+proc getClipboardText*(): string {.inline.} =
+  ## Get clipboard text content
+  result = $getClipboardTextPriv()
+
+proc getDroppedFiles*(): seq[string] =
+  ## Get dropped files names (memory should be freed)
+  var count = 0'i32
+  let dropfiles = getDroppedFilesPriv(count.addr)
+  result = cstringArrayToSeq(dropfiles, count)
+
+proc getGamepadName*(gamepad: int32): string {.inline.} =
+  ## Get gamepad internal name id
+  result = $getGamepadNamePriv(gamepad)
+
+proc loadModelAnimations*(fileName: string): seq[ModelAnimation] =
+  ## Load model animations from file
+  var len = 0'u32
+  let data = loadModelAnimationsPriv(fileName.cstring, len.addr)
+  if len <= 0:
+    raise newException(IOError, "No model animations loaded from " & filename)
+  result = newSeq[ModelAnimation](len.int)
+  copyMem(result[0].addr, data, len.int * sizeof(ModelAnimation))
+  #for i in 0..<len.int:
+    #result[i] = data[i]
+  memFree(data)
+
+proc loadWaveSamples*(wave: Wave): seq[float32] =
+  ## Load samples data from wave as a floats array
+  let data = loadWaveSamplesPriv(wave)
+  let len = int(wave.frameCount * wave.channels)
+  result = newSeq[float32](len)
+  copyMem(result[0].addr, data, len * sizeof(float32))
+  memFree(data)
+
+proc loadImageColors*(image: Image): seq[Color] =
+  ## Load color data from image as a Color array (RGBA - 32bit)
+  let data = loadImageColorsPriv(image)
+  let len = int(image.width * image.height)
+  result = newSeq[Color](len)
+  copyMem(result[0].addr, data, len * sizeof(Color))
+  memFree(data)
+
+proc loadImagePalette*(image: Image, maxPaletteSize: int32): seq[Color] =
+  ## Load colors palette from image as a Color array (RGBA - 32bit)
+  var len = 0'i32
+  let data = loadImagePalettePriv(image, maxPaletteSize, len.addr)
+  result = newSeq[Color](len.int)
+  copyMem(result[0].addr, data, len.int * sizeof(Color))
+  memFree(data)
+
+proc loadFontData*(fileData: openarray[uint8], fontSize: int32, fontChars: openarray[int32], `type`: FontType): seq[GlyphInfo] =
+  ## Load font data for further use
+  let data = loadFontDataPriv(cast[ptr UncheckedArray[uint8]](fileData), fileData.len.int32,
+      fontSize, cast[ptr UncheckedArray[int32]](fontChars), fontChars.len.int32, `type`)
+  result = newSeq[GlyphInfo](fontChars.len)
+  copyMem(result[0].addr, data, fontChars.len * sizeof(GlyphInfo))
+  memFree(data)
```

Comment:
I'm not sure we can allow ourself doing `newSeq` because it does extra allocation. Allocations are generally considered slow and this here will allocate twice the amount of memory and do twice the number of calls to allocate.

As I understand it is also impossible to make a sequence without addition allocations because of GC. Also, what would be a use case for sequences which we want to modify? There are length checks already in `raylib_fields.nim`. Here I assume we need sequences to dynamically modify them, but why? We should also take into considerations that operations on sequences will allocate memory uncontrollably which is sometimes undesirable for games.

Reply:
There's also a `View` type. I'm not sure about the progress on this. But it looks like a good option here. We can create a custom container which implements `toOpenArray` for each type as well as an option.

```nim
type GlyphInfoArray = object
  len: int,
  p: ptr UncheckedArray[GlyphInfo]

proc toOpenArray(obj: GlyphInfoArray) ...
```

Reply:
I will do that.

Reply:
Okay, great! Do you want to discuss these options on the specific examples? It could make sense to try several ways of doing this conversion before rewriting the whole file to conform to the one specific variant. There's at least `View` and custom container options. But we want to minimize the amount of work we are going to do. And possible automate it.

Regarding automation, it really seems that your version with generation from JSON is going to be better in terms of maintainability. But it's not entirely clear if this is the case.